### ISSUE TITLE
[MJAVADOC-145] If Javadoc is set to aggregate, the build fails inside a Maven plugin module

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#552]
**[Howard Lewis Ship](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hlship)** opened **[MJAVADOC-145](https://issues.apache.org/jira/browse/MJAVADOC-145?redirect=false)** and commented

If your project contains a Maven plugin, then it is impossible to build aggregated Javadoc.

As the output (attached) shows, Maven seems to recusively build (what's with all those "skipping" messages?).  When it reaches the plugin:

[INFO] Error during page generation

Embedded error: Error rendering Maven report: Error extracting plugin descriptor: 'Goal: component-report already exists in the plugin descriptor for prefix: tapestry-component-report
Existing implementation is: org.apache.tapestry.mojo.ComponentReport
Conflicting implementation is: org.apache.tapestry.mojo.ComponentReport'

I  can get by this with the -fn (fail never) option.

---

**Affects:** 2.3

**Attachments:**
- [ComponentReport.java](https://issues.apache.org/jira/secure/attachment/12716868/ComponentReport.java) (_16.01 kB_)
- [maven.out](https://issues.apache.org/jira/secure/attachment/12717080/maven.out) (_642.16 kB_)
- [pom.xml](https://issues.apache.org/jira/secure/attachment/12717058/pom.xml) (_4.05 kB_)
- [pom.xml](https://issues.apache.org/jira/secure/attachment/12716929/pom.xml) (_10.84 kB_)

**Issue Links:**
- [MJAVADOC-224](https://issues.apache.org/jira/browse/MJAVADOC-224) CLONE -If Javadoc is set to aggregate, the build fails inside a Maven plugin module
 (_**"duplicates"**_)
- [MJAVADOC-257](https://issues.apache.org/jira/browse/MJAVADOC-257) maven-javadoc-plugin 2.5, 2.6 and mojo: Error extracting plugin descriptor
 (_**"duplicates"**_)
- [MJAVADOC-168](https://issues.apache.org/jira/browse/MJAVADOC-168) Regression: 2.4-SNAPSHOT does not generate docs for generates sources if run outside a build

- [MJAVADOC-66](https://issues.apache.org/jira/browse/MJAVADOC-66) Aggregated Javadoc does not use module-level dependencies

**Comment by jira-importer** [comment]
**[Howard Lewis Ship](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hlship)** commented

Output from "mvn clean install site"

**Comment by jira-importer** [comment]
**[Howard Lewis Ship](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hlship)** commented

Project master POM.

**Comment by jira-importer** [comment]
**[Howard Lewis Ship](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hlship)** commented

POM for the tapestry-component-report module.

**Comment by jira-importer** [comment]
**[Howard Lewis Ship](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hlship)** commented

Mojo class.

**Comment by jira-importer** [comment]
**[Lukas Theussl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ltheussl)** commented

This seems to be a regression. We just had the same problem in doxia [1], it works with v 2.2.

[1] http://www.nabble.com/can%27t-build-site-tf4408106s177.html

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

It is due to `@phase` generate-sources and `@execute` phase="generate-sources" (introduced with MJAVADOC-66).
By default, the DefaultLifecycleExecutor uses the phase defined in the configuration, secondly the phase in the descriptor.

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

fixed in r574213 (removed `@execute` phase="generate-sources")

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

FYI I readded `@execute` phase="generate-sources" and `@execute` phase=generate-test-sources in r661899
I tried the tapestry build [1] with "mvn site -Pjavadoc" and all seems ok in 2.4 and 2.5-SNAPSHOT

[1] http://svn.apache.org/repos/asf/tapestry/tapestry5/trunk

**Comment by jira-importer** [comment]
**[Christian Schulte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=schulte77)** commented

Having

```xml
<reporting>
  <plugins>
    <plugin>
      <groupId>org.apache.maven.plugins</groupId>
      <artifactId>maven-javadoc-plugin</artifactId>
      <version>2.5</version>
      <reportSets>
        <reportSet>
          <id>javadoc</id>
          <inherited>true</inherited>
          <reports>
            <report>javadoc</report>
          </reports>
          <configuration>
            <aggregate>false</aggregate>
            <author>true</author>
            <charset>UTF-8</charset>
            <docencoding>UTF-8</docencoding>
            <encoding>UTF-8</encoding>
            <keywords>true</keywords>
            <doctitle>${pom.name} Specification ${pom.version}</doctitle>
            <linksource>true</linksource>
            <nodeprecated>true</nodeprecated>
            <quiet>false</quiet>
            <verbose>false</verbose>
            <serialwarn>true</serialwarn>
            <windowtitle>${pom.name} Specification ${pom.version}</windowtitle>
            <overview>${basedir}/src/main/javadoc/overview.html</overview>
            <show>public</show>
            <links>
              <link>http://java.sun.com/j2se/1.5.0/docs/api</link>
            </links>
          </configuration>
        </reportSet>
        <reportSet>
          <id>aggregate</id>
          <inherited>false</inherited>
          <reports>
            <report>aggregate</report>
          </reports>
          <configuration>
            <aggregate>true</aggregate>
            <breakiterator>true</breakiterator>
            <charset>UTF-8</charset>
            <docencoding>UTF-8</docencoding>
            <encoding>UTF-8</encoding>
            <keywords>false</keywords>
            <linksource>true</linksource>
            <serialwarn>true</serialwarn>
            <show>private</show>
            <doctitle>${pom.name} Specification ${pom.version}</doctitle>
            <windowtitle>${pom.name} Specification ${pom.version}</windowtitle>
          </configuration>
        </reportSet>
      </reportSets>
    </plugin>
```

in a parent pom from a multi-module project containing a mojo artifact defined like

```java
* @goal java-resources
* @phase generate-sources
* @requiresDependencyResolution compile
```

still produces a similar stacktrace for me

```java
[INFO] Trace
org.apache.maven.lifecycle.LifecycleExecutionException: Error extracting plugin descriptor: 'Goal: java-resources already exists in the plugin descriptor for prefix: some-prefix
Existing implementation is: some.package.JavaResourcesMojo
Conflicting implementation is: some.package.JavaResourcesMojo'
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:583)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalWithLifecycle(DefaultLifecycleExecutor.java:499)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.forkProjectLifecycle(DefaultLifecycleExecutor.java:924)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.forkLifecycle(DefaultLifecycleExecutor.java:767)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:549)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeStandaloneGoal(DefaultLifecycleExecutor.java:512)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoal(DefaultLifecycleExecutor.java:482)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalAndHandleFailures(DefaultLifecycleExecutor.java:330)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeTaskSegments(DefaultLifecycleExecutor.java:291)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:142)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:336)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:129)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:287)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:324)
        at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
        at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
        at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
        at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error extracting plugin descriptor: 'Goal: java-resources already exists in the plugin descriptor for prefix: some-prefix
Existing implementation is: some.package.JavaResourcesMojo
Conflicting implementation is: some.package.JavaResourcesMojo'
        at org.apache.maven.plugin.plugin.AbstractGeneratorMojo.execute(AbstractGeneratorMojo.java:158)
        at org.apache.maven.plugin.plugin.HelpGeneratorMojo.execute(HelpGeneratorMojo.java:63)
        at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:451)
        at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:558)
        ... 20 more
Caused by: org.apache.maven.plugin.descriptor.DuplicateMojoDescriptorException: Goal: java-resources already exists in the plugin descriptor for prefix: some-prefix
Existing implementation is: some.package.JavaResourcesMojo
Conflicting implementation is: some.package.JavaResourcesMojo
        at org.apache.maven.plugin.descriptor.PluginDescriptor.addMojo(PluginDescriptor.java:103)
        at org.apache.maven.tools.plugin.scanner.DefaultMojoScanner.populatePluginDescriptor(DefaultMojoScanner.java:110)
        at org.apache.maven.plugin.plugin.AbstractGeneratorMojo.execute(AbstractGeneratorMojo.java:146)
        ... 23 more

```

when executing

```java
mvn site:stage -DstagingDirectory=/tmp/staging -e
```

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

Christian, it is not a Javadoc plugin error so try to open a new issue in MNG with your test case. Thanks!

**Comment by jira-importer** [comment]
**[Martin Desruisseaux](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=desruisseaux)** commented

Is this bug really fixed? I'm using Maven 2.1 with `maven-javadoc-plugin` 2.5 in a project having a few Maven plugins, and still get "_Error extracting plugin descriptor: Goal already exists in the plugin descriptor_" when running `mvn clean install site`. The same command without the `site` goal works fine. Removing the aggregated javadoc configuration from the `<reporting>` section in the root `pom.xml` fixes the build as well.

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

Martin, it should be.
If you have an error, please create a new issue **with** a test project, I will have a look to.
